pkg/apply/applier.go

```diff
@@ -188,6 +185,65 @@ func splitInfos(infos []*resource.Info) ([]*resource.Info, []*resource.Info) {
 	return resources, groupingObjectTemplates
 }
 
+// buildTaskQueue takes the slice of infos and object identifiers, and
+// builds a queue of tasks that needs to be executed.
+func (a *Applier) buildTaskQueue(infos []*resource.Info, identifiers []object.ObjMetadata, eventChannel chan event.Event) chan taskrunner.Task {
```

Comment:
The event channel appears tightly coupled with the applier; I think we should make it a field in the applier struct instead of passing it around.

Reply:
The `eventChannel` is passed in on each call to `Run`, so it doesn't feel right to turn it into a field on the struct. Values that are used to configure the applier should be fields on the struct, while values passed into the applier through the `Run` function should not be fields. 
Also, I would like to keep the statements that create the channel and the statement that closes it as close together as possible.